### PR TITLE
docs: add documentation about `noPrimaryKey`

### PR DESCRIPTION
**Pull request by fzn0x**
Prevent creating a default PK attribute section.

### CODE REVIEW

**Comment by fzn0x** [comment]
This changes targets v7 only.

**Comment by fzn0x** [comment]
https://625d87150fc57260fa1a6178--sequelize-site.netlify.app/docs/v7/core-concepts/model-basics/#prevent-creating-a-default-pk-attribute

File: docs/core-concepts/model-basics.md
Comment (WikiRik) [inline docs/core-concepts/model-basics.md:299]: Maybe rephrase this so it's more clear that this will then happen only on a single model.

Something like: 'In case you only want to disable primary keys on a single model, you can define the field with `primaryKey: false`'

File: docs/core-concepts/model-basics.md
Comment (WikiRik) [inline docs/core-concepts/model-basics.md:283]:
```suggestion
By default, Sequelize automatically adds the primary key field `id` to every model when there is no primary key field defined. When there is an `id` field but not a primary key field Sequelize will throw an error. To prevent this you can use the option `noPrimaryKey` when creating the instance and set it to true.
```

Just a suggestion to make it more readable. Otherwise it is quite a long sentence

**Review by WikiRik — COMMENTED** [review]
I have some suggestions. Thanks again for your work on the PR in the main repo!

File: docs/core-concepts/model-basics.md
Comment (fzn0x) [reply docs/core-concepts/model-basics.md:299]: Thanks

**Review by WikiRik — DISMISSED** [review]

File: docs/core-concepts/model-basics.md
Comment (ephys) [inline docs/core-concepts/model-basics.md:283]: A small change because I would prefer to avoid using the word `field` and instead always refer to it as either `column` (when talking about sql) or `attribute` (when talking about js)

I would also propose to remove the part about sequelize throwing an error if `id` is added manually, but its `primaryKey` option is not configured for a few reasons:

- it's unrelated to `noPrimaryKey` (it works as a "hey, did you mean that?")
- the error it throws is clear enough
- it's not necessary for the user to know about it until they encounter the error, at which point the error will explain what is going on. They're never going to actively seek this information.

```suggestion
By default, Sequelize automatically adds the primary key attribute `id` to every model when there is no primary key has been defined manually. To prevent this you can use the option `noPrimaryKey` when creating the instance and set it to true.
```

File: docs/core-concepts/model-basics.md
Comment (ephys) [inline docs/core-concepts/model-basics.md:286]: `Sequelize.createSequelizeInstance` does not exist, it's a helper we use in our tests.

The way to create a sequelize instance is just `new Sequelize`

File: docs/core-concepts/model-basics.md
Comment (ephys) [inline docs/core-concepts/model-basics.md:299]: That's not true. Doing that will still add a primary key field called 'id'. It's only going to make `name` not a primary key, which would already be the case without the option.

This is how you disable adding the primary key using local options:

```js
const User = sequelize.define('User', {
  name: DataTypes.STRING,
}, {
  noPrimaryKey: true,
});
```

I would also suggest switching the order of explanations to start with the local options first, then the global ones (as we always do throughout the documentation)

**Review by ephys — CHANGES_REQUESTED** [review]

File: docs/core-concepts/model-basics.md
Comment (fzn0x) [reply docs/core-concepts/model-basics.md:283]: > A small change because I would prefer to avoid using the word `field` and instead always refer to it as either `column` (when talking about sql) or `attribute` (when talking about js)
> 
> I would also propose to remove the part about sequelize throwing an error if `id` is added manually, but its `primaryKey` option is not configured for a few reasons:
> 
> * it's unrelated to `noPrimaryKey` (it works as a "hey, did you mean that?")
> * the error it throws is clear enough
> * it's not necessary for the user to know about it until they encounter the error, at which point the error will explain what is going on. They're never going to actively seek this information.

Interesting explanation 😄 , I will continue with resolving the other tomorrow.

File: docs/core-concepts/model-basics.md
Comment (WikiRik) [reply docs/core-concepts/model-basics.md:286]: Woops, I missed that. Good one!

**Comment by fzn0x** [comment]
I think we are ready now, I waiting for your approval or comments @ephys @WikiRik

File: docs/core-concepts/model-basics.md
Comment (ephys) [inline docs/core-concepts/model-basics.md:283]: I left an extra `there is` during my previous review 😅 

Extra suggestion: reworded the second sentence because the example that follows doesn't show a creation of a Sequelize instance anymore but the creation of a model :)

```suggestion
By default, Sequelize automatically adds the primary key attribute `id` to every model when no primary key has been defined manually. To prevent this you can set the `noPrimaryKey` option to true when defining the model.
```

File: docs/core-concepts/model-basics.md
Comment (ephys) [inline docs/core-concepts/model-basics.md:293]: What do you think of this rewording? I'm proposing it because I'm afraid someone may think it means all primary keys will be disabled, even the ones added manually

```suggestion
In case you want to prevent the addition of the default foreign key for every model:
```

Comment (fzn0x) [reply docs/core-concepts/model-basics.md:293]: > 

I think it's nice!

Comment (fzn0x) [reply docs/core-concepts/model-basics.md]: It's my mistake to not aware about it

File: docs/core-concepts/model-basics.md
Comment (ephys) [inline docs/core-concepts/model-basics.md:293]: Damn it, this is me again

```suggestion
In case you want to prevent the addition of the default primary key for every model:
```

**Review by ephys — CHANGES_REQUESTED** [review]

**Review by ephys — APPROVED** [review]
Looks good, I have no more notes :)

**Comment by fzn0x** [comment]
Hi @WikiRik How about you? :) I need two approval for this

**Comment by WikiRik** [comment]
Looks good!

**Pull request merged**
